modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM623AddDocumentationToAnAPIWithDocTypePublicForumThroughPublisherRestAPITestCase.java

```diff
@@ -112,36 +106,35 @@ public void testAddDocumentToAnAPIPublicInline() throws Exception {
-        JSONObject apiResponse = new JSONObject(apiCreationResponse.getData());
-        assertEquals(apiCreationResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                "Response Code miss matched when creating the API");
-        assertFalse(apiResponse.getBoolean("error"), apiName + "is not created as expected");
-
+        APIDTO apiCreationResponse = restAPIPublisher.addAPI(apiCreationRequestBean);
+        String status = apiCreationResponse.getLifeCycleStatus();
+        apiId = apiCreationResponse.getId();
+        assertTrue(APILifeCycleState.CREATED.getState().equalsIgnoreCase(status), "Status of the " + apiName +
+                "is not a valid status");
         //Check availability of the API in publisher
-        HttpResponse apiResponsePublisher = apiPublisher.getAPI(apiName, apiProvider, apiVersion);
-        JSONObject jsonObject = new JSONObject(apiResponsePublisher.getData());
-        JSONObject apiObject = new JSONObject(jsonObject.getString("api"));
-        assertFalse(jsonObject.getBoolean("error"), apiName + " is not visible in publisher");
-        assertTrue(apiObject.getString("name").equals(apiName),
-                apiName + " is not visible in publisher");
-        assertTrue(apiObject.getString("status").equals("CREATED"),
-                "Status of the " + apiName + "is not a valid status");
-
+        HttpResponse apiResponsePublisher = restAPIPublisher.getAPI(apiId);
+        assertEquals(apiResponsePublisher.getResponseCode(), Response.Status.OK.getStatusCode(), apiName +
+                " is not visible in publisher");
         //Add Documentation to an API "APIM623PublisherTest" -  public forum | Url
-        HttpResponse docResponse = apiPublisher.addDocument
-                (apiName, apiVersion, apiProvider, docName, docType, sourceType, docUrl,
-                        "Testing", "", "", null);
-        JSONObject jsonObjectDoc1 = new JSONObject(docResponse.getData());
-        assertFalse(jsonObjectDoc1.getBoolean("error"), "Error when adding document with source " +
-                "Url to the API");
+        DocumentDTO documentDTO = new DocumentDTO();
+        documentDTO.setName(docName);
+        documentDTO.setType(DocumentDTO.TypeEnum.PUBLIC_FORUM);
+        documentDTO.setSourceType(DocumentDTO.SourceTypeEnum.URL);
+        documentDTO.setSummary(summary);
+        documentDTO.setVisibility(DocumentDTO.VisibilityEnum.API_LEVEL);
+        documentDTO.setSourceUrl(docUrl);
+        HttpResponse documentationResponse = restAPIPublisher.addDocument(apiId, documentDTO);
+        assertEquals(documentationResponse.getResponseCode(), 200, "Error while add documentation to API");
+        forumTypeDocumentId = documentationResponse.getData();
+        HttpResponse docRemoveResponse = restAPIPublisher.removeDocumentation(apiId, forumTypeDocumentId);
+        assertEquals(Response.Status.OK.getStatusCode(), docRemoveResponse.getResponseCode(), "Error when removing" +
```

Comment:
According to the [assertEquals](https://javadoc.io/doc/org.testng/testng/6.8.17/org/testng/Assert.html#assertEquals(boolean,%20boolean)) method documentation, the First argument should be the actual value and the second argument should be the expected value, So I think here, arguments should be swapped.

Reply:
fixed in 
51770a5